CRM/Speakcivi/Logic/Cache/SpeakoutCampaign.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+class CRM_Speakcivi_Logic_Cache_SpeakoutCampaign extends CRM_Speakcivi_Logic_Cache
+{
+
+    const CACHE_PREFIX = '-speakout-';
+
+    public static function getCampaign($speakoutDomain, $speakoutCampaignId)
+    {
+        if ($speakout_campaign = self::get(
+            self::CACHE_PREFIX,
+            "$speakoutDomain:$speakoutCampaignId"
+        )) {
+            return $speakout_campaign;
+        }
+
+        $url = "https://$speakoutDomain/api/v1/campaigns/$speakoutCampaignId";
+        $user = CIVICRM_SPEAKOUT_USERS[$speakoutDomain];
+        $auth = $user['email'] . ':' . $user['password'];
+        $speakout_campaign = (array)json_decode(fetchSpeakoutContent($url, $auth));
+
+        self::set(self::CACHE_PREFIX, "$speakoutDomain:$speakoutCampaignId", $speakout_campaign, 300); # TODO: Set in configuration
```

Comment:
CRM_Speakcivi_Logic_Cache::set() method has 3 arguments, not 4. I mean "300".

Reply:
> CRM_Speakcivi_Logic_Cache::set() method has 3 arguments, not 4. I mean "300".

Oh right - thanks. I was looking at the Civi::ArrayCache code which has a $ttl argument. I can update this code to use one for now, but should we use the one that's built into the cache?

Reply:
But clearly my testing isn't hitting this code! Thank you!